pandas/core/resample.py

```diff
@@ -1404,15 +1404,18 @@ def __init__(
         self.fill_method = fill_method
         self.limit = limit
 
-        if origin in ("epoch", "start", "start_day"):
+        if origin in ("epoch", "start", "start_day", "end"):
+            if origin == "end" and self.closed == "left":
```

Comment:
Do you know if we have a similar constraint/check for if origin == "start" and self.closed == "right":

Reply:
No, now `start` use `left` by default (except for ‘M’, ‘A’, ‘Q’, ‘BM’, ‘BA’, ‘BQ’, and ‘W’).
Thus I will make a change to allow 'left' yet still use `right` as default setting for `end`, this may be more consistent.